tests/port_dpb.py

```diff
@@ -203,3 +203,92 @@ def verify_asic_db(self):
             assert(fvs_dict['SAI_PORT_ATTR_HW_LANE_LIST'] == self.get_lanes_asic_db_str())
         assert(fvs_dict['SAI_PORT_ATTR_SPEED'] == str(self.get_speed()))
 
+class DPB():
+    def breakin(self, dvs, port_names):
+        child_ports = []
+        for pname in port_names:
+            cp = Port(dvs, pname)
+            cp.sync_from_config_db()
+            child_ports.append(cp)
+
+        for cp in child_ports:
+            cp.delete_from_config_db()
+            # TBD, need vs lib to support removing hostif 
+            #dvs.runcmd("ip link delete " + cp.get_name())
+        #print "Deleted child ports:%s from config DB"%port_names
+
+        time.sleep(6)
+
+        for cp in child_ports:
+            assert(cp.exists_in_config_db() == False)
+        for cp in child_ports:
+            assert(cp.exists_in_app_db() == False)
+        for cp in child_ports:
+            assert(cp.exists_in_asic_db() == False)
+        #print "Verified child ports are deleted from all DBs"
+
+        p = Port(dvs)
+        p.port_merge(child_ports)
+        p.write_to_config_db()
+        #print "Added port:%s to config DB"%p.get_name()
+        time.sleep(2)
+
+        p.verify_config_db()
+        #print "Config DB verification passed!"
+
+        p.verify_app_db()
+        #print "Application DB verification passed!"
+
+        p.verify_asic_db()
+        #print "ASIC DB verification passed!"
+
+    def breakout(self, dvs, port_name, num_child_ports):
+
+        p = Port(dvs, port_name)
+        p.sync_from_config_db()
+
+        # Delete port from config DB and kernel
+        p.delete_from_config_db()
+        # TBD, need vs lib to support hostif removal
+        #dvs.runcmd("ip link delete " + p.get_name())
+        #print "Deleted port:%s from config DB"%port_name
+        time.sleep(6)
+
+        # Verify port is deleted from all DBs
+        assert(p.exists_in_config_db() == False)
+        assert(p.exists_in_app_db() == False)
+        assert(p.exists_in_asic_db() == False)
+
+        # Create child ports and write to config DB
+        child_ports = p.port_split(num_child_ports)
+        child_port_names = []
+        for cp in child_ports:
+            cp.write_to_config_db()
+            child_port_names.append(cp.get_name())
+        #print "Added child ports:%s to config DB"%child_port_names
+        time.sleep(6)
```

Comment:
We need understand the delay required here.  In general, we shouldn't need wait this long.

Reply:
Sure. I will create a ticket for it and work on later..